### DISCUSSION TITLE
Jobs run concurrently even on using DisallowConcurrentExecution attribute on job class and unique job key

### DISCUSSION THREAD

**Discussion by kibo-raghu** [quartznet/quartznet#937] [General]
Hello,
I am using Quartz.net 3.0.7 and ADO.Net job store(SQL server) to set up the scheduler. Few jobs on my scheduler should not be allowed to run concurrently . I used DisallowConcurrentExecution attribute on job class and same job key(Same job name and job group) if the job is triggered multiple times. 

I can see a single row getting generated on QRTZ_SIMPLE_TRIGGERS and QRTZ_TRIGGERS but multiple rows on QRTZ_FIRED_TRIGGERS with the status "Executing". I can see both the rows have similar data on job name and job group columns. I am not able to figure out why there are multiple instances running at the same time.I am attaching the screenshots of database tables for reference. Please help me with the issue..
<img width="1071" alt="Screen Shot 2020-08-19 at 3 34 00 PM" src="https://user-images.githubusercontent.com/68894348/90632334-dc009a00-e241-11ea-98a1-4ad0e5d54bef.png">
<img width="1075" alt="Screen Shot 2020-08-19 at 3 34 23 PM" src="https://user-images.githubusercontent.com/68894348/90632347-e1f67b00-e241-11ea-83b1-4dff971016f3.png">

**Comment by lahma** [comment]
There is only single instance of Quartz running, no multiple servers?

**Comment by kibo-raghu** [comment]
Yes.Only single instance of Quartz running in the docker container.

**Comment by kibo-raghu** [comment]
So..Do you think scaling the app to multiple containers would cause concurrency issues?

**Comment by lahma** [comment]
Of course. You need to configure all schedulers as clustered and ensure time-sync if there is at any point multiple schedulers accessing same database (having same scheduler name).

**Comment by kibo-raghu** [comment]
I do have a single container currently and still see jobs run concurrently(After specifying the DisallowConcurrentExecution and same job key). What else can I do to mitigate this?

**Reply by lahma** [reply]
Well I would argue that there's not bug on Quartz's side. Probably something is interfering in your setup. Hard to give advice with this background information.
